lib/web/join_tokens.go

```diff
@@ -299,7 +299,13 @@ func getJoinScript(ctx context.Context, settings scriptSettings, m nodeAPIGetter
 	}
 
 	version := proxyServers[0].GetTeleportVersion()
-	hostname, portStr, err := utils.SplitHostPort(proxyServers[0].GetPublicAddr())
+
+	publicAddr := proxyServers[0].GetPublicAddr()
+	if publicAddr == "" {
+		publicAddr = defaults.ProxyWebListenAddr().Addr
```

Comment:
Should `GetPublicAddr` maybe return this value by default if the public address is empty?